roles/docker-compose/tasks/install_docker_compose.yml

```diff
@@ -0,0 +1,31 @@
+---
+
+- name: Install dependencies
+  apt:
+    name: '{{ packages }}'
+    update_cache: yes
+    cache_valid_time: 3600
+  vars:
+    packages:
+    - build-essential
+    - libssl-dev
+    - python-dev
+    - python-pip
+  become: yes
+  become_method: sudo
+
+- name: Install pipsi
+  pip:
+    name: '{{ item }}'
+    state: forcereinstall
+  with_items:
+    - pipsi
+  executable: pip3
+  become: yes
+  become_method: sudo
+
+- name: Install docker-compose
+  command:
+    pipsi install docker-compose==1.22.0
```

Comment:
While this works, the recommend approach is to write a module wrapping this.

Reply:
How do I write a module to wrap this?

Reply:
It's relatively simple. You may look up some of the existing modules:
* https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/portage.py
* https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/language/pip.py
* https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/apt.py

Personally, I prefer OOP approach (wrapping that into a class with entry point and custom exception machinery), like in:
https://github.com/webknjaz/ansible-gentoo-laptop/blob/master/library/eix.py
(yes, if you put it into `library/` folder in this repo, you'll be able to use it by its name)

There's `ansible.module_utils.basic.AnsibleModule` helper you can import if you're writing it in Python.

Some docs on this are here: https://docs.ansible.com/ansible/latest/dev_guide/developing_modules.html

*TL;DR* your module accepts some JSON via stdin and responds with JSON to stdout. That's how ansible core is talking to modules.
Initialize `AnsibleModule` helper containing schema for accepted args. Then this instance has `.exit_json()` and `.fail_json()` to return some result. Other parts of flow are up to you, in general you'd run some tool you're wrapping via subprocess and then check the result.
Keep in mind to use Python syntax compatible with the interpreter on target hosts, best thing to do is useing subset which is common across Python 2 and Python 3 + also have `__future__` imports + `__metaclass__ = type` hack (makes classes in Python 2 new-style by default, no need to inherit from object explicitly).

Reply:
nice! thanks. you should make a blog post with just this :smile:

Reply:
I think, you can find some existing blog posts in the wild :)